lib/src/components/NxToast/NxToast.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import React, { useState, useContext, useEffect } from 'react';
+import classnames from 'classnames';
+import FocusContext from './contexts';
+
+import { nxToastPropTypes, NxToastProps, ToastModel } from './types';
+
+export { ToastModel };
+
+const NxToast = (props: NxToastProps) => {
+  const { onClose, children } = props;
+
+  const [isClosing, setIsClosing] = useState(false);
+
+  const focusedToast = useContext(FocusContext);
+
+  useEffect(()=>{
+    focusedToast?.onToastClosing();
+    return () => { Promise.resolve().then(() => focusedToast?.onToastClosing()); };
+  }, []);
+
+  const handleClose = () => {
+    if (!isClosing) {
+      setIsClosing(true);
+    }
+  };
+
+  const handleAnimationEnd = () => {
+    if (isClosing) {
+      onClose();
+    }
+  };
+
+  const validChild = React.Children.only(children);
+  const childrenWithProps = React.isValidElement(children) ?
+    React.cloneElement(validChild, { onClose: handleClose }) : null;
+
+  if (!React.isValidElement(children)) {
+    throw new TypeError('NxToast only accepts one NxAlert as a valid child element');
+  } 
```

Comment:
```suggestion
  if (!React.isValidElement(children)) {
    throw new TypeError('NxToast only accepts one NxAlert as a valid child element');
  } 
  
  const childrenWithProps = React.cloneElement(validChild, { onClose: handleClose });
```

Reply:
Oh, actually it turns out `React.Children.only` already asserts that the child is a React element (in addition to asserting that there's only one child) so the extra `isValidElement` check is unneeded entirely.

Reply:
> Oh, actually it turns out `React.Children.only` already asserts that the child is a React element (in addition to asserting that there's only one child) so the extra `isValidElement` check is unneeded entirely.

So then revert it back to as it was prior to adding the new TypeError?

Reply:
> So then revert it back to as it was prior to adding the new TypeError?

No, no need for the conditional

Reply:
> No, no need for the conditional

If typescript complains, you might need a cast.  The types for `only` don't seem right to me.

Reply:
> > Oh, actually it turns out React.Children.only already asserts that the child is a React element (in addition to asserting that there's only one child) so the extra isValidElement check is unneeded entirely.
> 
> If typescript complains, you might need a cast. The types for `only` don't seem right to me.

Implemented here: [a10a2d5](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/a10a2d51bd38d9fe3e6b555910ac9271eb3f99cb) . No typescript errors appeared

Reply:
>> Oh, actually it turns out `React.Children.only` already asserts that the child is a React element (in addition to asserting that there's only one child) so the extra `isValidElement` check is unneeded entirely.
>
>If typescript complains, you might need a cast. The types for only don't seem right to me.

Removing the conditional implemented here: [66d761e](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/66d761ee272944bd693903b37d5ba9eefec45b23) . Typescript didn't give any errors

Reply:
> Removing the conditional implemented here: [66d761e](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/66d761ee272944bd693903b37d5ba9eefec45b23) . Typescript didn't give any errors

You still have the `throw` and its conditional.  My point was to get rid of that (and the ternary that you got rid of as well), no need for any conditional logic at all in this stretch of code.

Reply:
> You still have the `throw` and its conditional. My point was to get rid of that (and the ternary that you got rid of as well), no need for any conditional logic at all in this stretch of code.

All removed here: [08c6e2e](https://github.com/sonatype/sonatype-react-shared-components/pull/688/commits/08c6e2e7e13ca229d01d3ac0a952800fe8655433)